feature/experimental/gribi/ate_tests/route_removal_during_failover_test/route_removal_during_failover_test.go

```diff
@@ -0,0 +1,565 @@
+				WithElectionID(args.electionID.Low, args.electionID.High))
+	}
+
+	time.Sleep(time.Minute)
+	virtualVIPs := createIPv4Entries("198.18.196.1/22")
+
+	for ip := range virtualVIPs {
+		args.client.Modify().AddEntry(t,
+			fluent.IPv4Entry().
+				WithPrefix(virtualVIPs[ip]+"/32").
+				WithNetworkInstance(*deviations.DefaultNetworkInstance).
+				WithNextHopGroup(uint64(nhgID)).
+				WithElectionID(args.electionID.Low, args.electionID.High))
+	}
+	if err := awaitTimeout(args.ctx, args.client, t, time.Minute); err != nil {
+		t.Fatalf("Could not program entries via clientA, got err: %v", err)
+	}
+
+	for ip := range virtualVIPs {
+		chk.HasResult(t, args.client.Results(t),
+			fluent.OperationResult().
+				WithIPv4Operation(virtualVIPs[ip]+"/32").
+				WithOperationType(constants.Add).
+				WithProgrammingResult(fluent.InstalledInFIB).
+				AsResult(),
+			chk.IgnoreOperationID(),
+		)
+	}
+}
+
+// pushConfig pushes the configuration generated by this
+// struct to the device using gNMI SetReplace.
+func pushConfig(t *testing.T, dut *ondatra.DUTDevice, dutPort *ondatra.Port, d *oc.Root) {
+	t.Helper()
+
+	iname := dutPort.Name()
+	i := d.GetOrCreateInterface(iname)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(iname).Config(), i)
+}
+
+// configureInterfaceDUT configures a single DUT layer 2 port.
+func configureInterfaceDUT(t *testing.T, dutPort *ondatra.Port, d *oc.Root, desc string) {
+	t.Helper()
+
+	i := d.GetOrCreateInterface(dutPort.Name())
+	i.Description = ygot.String(desc)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+	t.Logf("DUT port %s configured", dutPort)
+}
+
+// generateSubIntfPair takes the number of subInterfaces, dut,ate,ports and Ixia topology.
+// It configures ATE/DUT SubInterfaces on the target device
+// It returns a slice of the corresponding ATE IPAddresses.
+func generateSubIntfPair(t *testing.T, dut *ondatra.DUTDevice, dutPort *ondatra.Port, ate *ondatra.ATEDevice, atePort *ondatra.Port, top *ondatra.ATETopology, d *oc.Root) []string {
+	nextHops := []string{}
+	nextHopCount := 63 // nextHopCount specifies number of nextHop IPs needed.
+	for i := 0; i <= nextHopCount; i++ {
+		vlanID := uint16(i)
+		if *deviations.NoMixOfTaggedAndUntaggedSubinterfaces {
+			vlanID = uint16(i) + 1
+		}
+		name := fmt.Sprintf(`dst%d`, i)
+		Index := uint32(i)
+		ateIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 1))
+		dutIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 2))
+		configureSubinterfaceDUT(t, d, dutPort, Index, vlanID, dutIPv4)
+		configureATE(t, top, atePort, name, vlanID, dutIPv4, ateIPv4+"/30")
+		nextHops = append(nextHops, ateIPv4)
+	}
+	configureInterfaceDUT(t, dutPort, d, "dst")
+	pushConfig(t, dut, dutPort, d)
+	return nextHops
+}
+
+// configureSubinterfaceDUT configures a single DUT layer 3 sub-interface.
+func configureSubinterfaceDUT(t *testing.T, d *oc.Root, dutPort *ondatra.Port, index uint32, vlanID uint16, dutIPv4 string) {
```

Comment:
looks like the t parameter is unused

Reply:
I have added t.Helper() here so t will be used.